File: tests/fixtures/python_project/calculator.py
"""Calculator module with basic arithmetic operations"""

from utils.helpers import validate_input, format_result


def add(a, b):
    """Adds two numbers together
    
    Args:
        a: First number
        b: Second number
        
    Returns:
        The sum of a and b
    """
    return a + b


def subtract(a, b):
    """Subtracts b from a"""
    return a - b


def multiply(a, b):
    """Multiplies two numbers"""
    return a * b


def divide(a, b):
    """Divides a by b, returns None if b is zero"""
    if b == 0:
        return None
    return a / b


def _private_helper(x):
    """Private helper function"""
    return x >= 0


def apply_operation(a, b, operation):
    """Applies a custom operation to two numbers
    
    Args:
        a: First number
        b: Second number
        operation: A function that takes two numbers
        
    Returns:
        Result of the operation
    """
    result = operation(a, b)
    
    # Nested function
    def formatter(x):
        return f"Result is: {x}"
    
    print(formatter(result))
    return result



class Calculator:
    """A simple calculator class
    
    This class maintains state for calculator operations.
    """
    
    def __init__(self, initial_value=0):
        """Creates a new Calculator
        
        Args:
            initial_value: Starting value (default: 0)
        """
        self.value = initial_value
        self._history = []
    
    def add(self, n):
        """Adds a number to the current value"""
        self.value += n
        self._history.append(f"add {n}")
        return self.value
    
    def subtract(self, n):
        """Subtracts a number from the current value"""
        self.value -= n
        self._history.append(f"subtract {n}")
        return self.value
    
    def reset(self):
        """Resets the calculator to zero"""
        self.value = 0
        self._history.clear()
    
    def get_history(self):
        """Gets the operation history"""
        return self._history.copy()
    
    def _log_operation(self, op):
        """Private helper method"""
        self._history.append(op)
    
    def __str__(self):
        return f"Calculator(value: {self.value})"


class Point:
    """A point structure for testing"""
    
    def __init__(self, x, y):
        """Creates a new point"""
        self.x = x
        self.y = y
    
    @staticmethod
    def origin():
        """Creates a point at the origin"""
        return Point(0, 0)
    
    def distance_from_origin(self):
        """Calculates distance from origin"""
        return (self.x ** 2 + self.y ** 2) ** 0.5
    
    def distance_to(self, other):
        """Calculates distance to another point"""
        dx = self.x - other.x
        dy = self.y - other.y
        return (dx ** 2 + dy ** 2) ** 0.5
    
    def translate(self, dx, dy):
        """Translates the point by the given offset"""
        self.x += dx
        self.y += dy
    
    def translated(self, dx, dy):
        """Returns a new point translated by the given offset"""
        return Point(self.x + dx, self.y + dy)


class LineSegment:
    """A line segment between two points"""
    
    def __init__(self, start, end):
        """Creates a new line segment
        
        Args:
            start: Starting point
            end: Ending point
        """
        self.start = start
        self.end = end
    
    def length(self):
        """Calculates the length of the line segment"""
        return self.start.distance_to(self.end)
    
    def midpoint(self):
        """Calculates the midpoint of the line segment"""
        mid_x = (self.start.x + self.end.x) / 2
        mid_y = (self.start.y + self.end.y) / 2
        return Point(mid_x, mid_y)

File: tests/fixtures/javascript_project/calculator.js
/**
 * Calculator module with basic arithmetic operations
 */

const { validateInput } = require('./utils/helpers');

/**
 * Adds two numbers together
 * @param {number} a - First number
 * @param {number} b - Second number
 * @returns {number} The sum
 */
function add(a, b) {
    return a + b;
}

/**
 * Subtracts b from a
 * @param {number} a - The minuend
 * @param {number} b - The subtrahend
 * @returns {number} The difference
 */
function subtract(a, b) {
    return a - b;
}

/**
 * Multiplies two numbers
 * @param {number} a - First number
 * @param {number} b - Second number
 * @returns {number} The product
 */
function multiply(a, b) {
    return a * b;
}

/**
 * Divides a by b, returns null if b is zero
 * @param {number} a - The dividend
 * @param {number} b - The divisor
 * @returns {number|null} The quotient or null
 */
function divide(a, b) {
    if (b === 0) {
        return null;
    }
    return a / b;
}

function applyOperation(a, b, operation) {
    
    
    
    
    const formatter = (x) => `Result is: ${x}`;
    const result = operation(a, b);
    
    console.log(formatter(result));
    return result;
}



/**
 * A simple calculator class
 * 
 * This class maintains state for calculator operations.
 */
class Calculator {
    /**
     * Creates a new Calculator
     * @param {number} initialValue - Starting value (default: 0)
     */
    constructor(initialValue = 0) {
        this.value = initialValue;
        this._history = [];
    }
    
    
    add(n) {
        this.value += n;
        this._history.push(`add ${n}`);
        return this.value;
    }
    
    /**
     * Subtracts a number from the current value
     * @param {number} n - Number to subtract
     * @returns {number} The new value
     */
    subtract(n) {
        this.value -= n;
        this._history.push(`subtract ${n}`);
        return this.value;
    }
    
    /**
     * Resets the calculator to zero
     */
    reset() {
        this.value = 0;
        this._history = [];
    }
    
    /**
     * Gets the operation history
     * @returns {Array<string>} Copy of the history
     */
    getHistory() {
        return [...this._history];
    }
    
    /**
     * Private helper method
     * @private
     */
    _logOperation(op) {
        this._history.push(op);
    }
    
    toString() {
        return `Calculator(value: ${this.value})`;
    }
}

/**
 * A point class for testing
 */
class Point {
    /**
     * Creates a new point
     * @param {number} x - X coordinate
     * @param {number} y - Y coordinate
     */
    constructor(x, y) {
        this.x = x;
        this.y = y;
    }
    
    /**
     * Creates a point at the origin
     * @static
     * @returns {Point} A point at (0, 0)
     */
    static origin() {
        return new Point(0, 0);
    }
    
    /**
     * Calculates distance from origin
     * @returns {number} The distance
     */
    distanceFromOrigin() {
        return Math.sqrt(this.x ** 2 + this.y ** 2);
    }
    
    /**
     * Calculates distance to another point
     * @param {Point} other - The other point
     * @returns {number} The distance
     */
    distanceTo(other) {
        const dx = this.x - other.x;
        const dy = this.y - other.y;
        return Math.sqrt(dx ** 2 + dy ** 2);
    }
    
    /**
     * Translates the point by the given offset
     * @param {number} dx - X offset
     * @param {number} dy - Y offset
     */
    translate(dx, dy) {
        this.x += dx;
        this.y += dy;
    }
    
    /**
     * Returns a new point translated by the given offset
     * @param {number} dx - X offset
     * @param {number} dy - Y offset
     * @returns {Point} A new translated point
     */
    translated(dx, dy) {
        return new Point(this.x + dx, this.y + dy);
    }
}

/**
 * A line segment between two points
 */
class LineSegment {
    /**
     * Creates a new line segment
     * @param {Point} start - Starting point
     * @param {Point} end - Ending point
     */
    constructor(start, end) {
        this.start = start;
        this.end = end;
    }
    
    /**
     * Calculates the length of the line segment
     * @returns {number} The length
     */
    length() {
        return this.start.distanceTo(this.end);
    }
    
    /**
     * Calculates the midpoint of the line segment
     * @returns {Point} The midpoint
     */
    midpoint() {
        const midX = (this.start.x + this.end.x) / 2;
        const midY = (this.start.y + this.end.y) / 2;
        return new Point(midX, midY);
    }
}

module.exports = {
    add,
    subtract,
    multiply,
    divide,
    applyOperation,
    complexOperation,
    pointDistance,
    Calculator,
    Point,
    LineSegment
};
